### PR TITLE
Ntree add test

### PR DESCRIPTION
**Pull request by ineiti**
Tests for NTree - will also add tests for `protocols/manage`.

### CODE REVIEW

**Comment by ineiti** [comment]
Also added tests for the `protocols/manage`-protocols. Couldn't resist to rewrite the broadcast protocol which sent around too many messages.

**Comment by Liamsi** [comment]
Looks OK

File: protocols/manage/broadcast.go [hunk 1]
```diff
@@ -14,146 +16,82 @@ func init() {
...
+	dbg.Lvl3(b.Info(), "Received message from", msg.TreeNode.String())
+	if msg.TreeNode.Id == b.Root().Id {
+		dbg.Lvl3(b.Info(), "Contacting everybody")
+		// Connect to all nodes that are later in the TreeNodeList, but only if
+		// the message comes from root
+		for _, tn := range b.Tree().List()[b.tnIndex+1:] {
+			dbg.Lvl3("Connecting to", tn.String())
+			err := b.SendTo(tn, &ContactNodes{})
+			if err != nil {
+				return err
```
Comment (nikkolasg) [inline protocols/manage/broadcast.go]: In fact, having a handler does not necessarily have to return an error. `RegisterHandler` takes an interface, and we don't check for error, so a `dbg.Error` could do the trick.

**Comment by nikkolasg** [comment]
Except comment, OK for me too.

**Comment by ineiti** [comment]
Removed error-handling (sorry Liamsi...) but added a note to good old #189

**Comment by ineiti** [comment]
Better now - there was a handler that was also used by `Start`, so it should return an error.

**Pull request merged**